packages/autorest.python/autorest/codegen/models/parameter.py

```diff
@@ -278,6 +280,25 @@ def has_json_model_type(self) -> bool:
             return self.type.target_model_subtype((JSONModelType,)) is not None
         return isinstance(self.type, JSONModelType)
 
+    @property
```

Comment:
also we can actually support `data` and `files` entered, and pass them as `HttpRequest(data=data, files=files)` once this pr gets merged and released https://github.com/Azure/azure-sdk-for-python/pull/34021

Reply:
Before https://github.com/Azure/autorest.python/pull/2380#discussion_r1467089995, it passes `data` and `files`. Now the code merge all part into files and it works. SDK users will not be influenced no matter SDK merge all in `files` or pass `data`/`files` so I think current logic is OK.